Google.PowerShell/Storage/GcsObject.cs

```diff
@@ -832,10 +841,25 @@ protected override void ProcessRecord()
             using (var writer = new FileStream(qualifiedPath, FileMode.Create))
             {
                 var result = downloader.Download(uri, writer);
-                if (result.Status == DownloadStatus.Failed || result.Exception != null)
+                CheckForError(result);
+            }
+        }
+
+        private void CheckForError(IDownloadProgress downloadProgress)
+        {
+            if (downloadProgress.Status == DownloadStatus.Failed || downloadProgress.Exception != null)
+            {
+                string exceptionMessage = downloadProgress.Exception.Message;
+                if (exceptionMessage != null && exceptionMessage.IndexOf("Not Found", System.StringComparison.OrdinalIgnoreCase) >= 0)
```

Comment:
Is this a safe check? Shouldn't we check and error code? 

exceptionMessage.IndexOf("Not Found"